BrowserServicesKit/Sources/Subscription/Managers/SubscriptionManagerV2.swift

```diff
@@ -367,18 +377,23 @@ public final class DefaultSubscriptionManagerV2: SubscriptionManagerV2 {
             }
 
             return resultTokenContainer
-        } catch OAuthClientError.refreshTokenExpired {
-            do { return try await attemptTokenRecovery() } catch { throw error }
+        } catch OAuthClientError.refreshTokenExpired, OAuthClientError.invalidTokenRequest {
+            do {
+                return try await attemptTokenRecovery()
+            } catch {
+                throw error
+            }
         } catch {
             throw SubscriptionManagerError.tokenUnavailable(error: error)
         }
     }
 
     func attemptTokenRecovery() async throws -> TokenContainer {
         Logger.subscription.log("The refresh token is expired, attempting subscription recovery...")
+        pixelHandler(.deadToken)
         await signOut(notifyUI: false)
         do {
-            try await autoRecoveryHandler()
+            try await tokenRecoveryHandler?()
             return try await getTokenContainer(policy: .local)
         } catch {
             throw SubscriptionManagerError.tokenUnRefreshable
```

Comment:
Related to earlier comment, I see more `SubscriptionManagerError.tokenUnRefreshable` to be `SubscriptionManagerError.tokenRefreshFailed`. As it not necessary means that it unrefreshable but the recovery failed due to one of the reasons.